# Carregando arquivo de dados

In [35]:
with open('corretor-master/artigos.txt', 'r', errors = 'ignore', encoding = 'utf-8') as f:
    artigos = f.read()

In [36]:
print(artigos[:500])




imagem 

Temos a seguinte classe que representa um usuário no nosso sistema:

java

Para salvar um novo usuário, várias validações são feitas, como por exemplo: Ver se o nome só contém letras, [**o CPF só números**] e ver se o usuário possui no mínimo 18 anos. Veja o método que faz essa validação:

java 

Suponha agora que eu tenha outra classe, a classe `Produto`, que contém um atributo nome e eu quero fazer a mesma validação que fiz para o nome do usuário: Ver se só contém letras. E aí? Vou


In [37]:
len(artigos)

2605046

In [38]:
texto_exemplo = "Olá, tudo bem?"
tokens = texto_exemplo.split()

In [39]:
tokens

['Olá,', 'tudo', 'bem?']

In [40]:
len(tokens)

3

# NLTK separando os tokens

In [41]:
import nltk

In [42]:
nltk.download('punkt')

[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\Dorivan\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!


True

In [43]:
# Separa as palavras dos caracteres
palavras_separadas = nltk.tokenize.word_tokenize(texto_exemplo)

In [44]:
palavras_separadas

['Olá', ',', 'tudo', 'bem', '?']

In [45]:
len(palavras_separadas)

5

In [46]:
'palavra'.isalpha()

True

In [47]:
# Percorre todos os dados do parâmetro, e atribui a uma lista apenas as palavras que são Alpha, que conteém letras
def separa_palavras(lista_tokens):
    lista_palavras = []
    for token in lista_tokens:
        if token.isalpha():
            lista_palavras.append(token)
    return lista_palavras

In [48]:
# Testando a função
separa_palavras(palavras_separadas)

['Olá', 'tudo', 'bem']

In [52]:
# Executando a função com os dados do 'artigo'
# variável que recebe as funcões de tokenização da base 'artigos'
lista_tokens = nltk.tokenize.word_tokenize(artigos)

# Executa a função com a variável que recebeu a tokenização
lista_palavras = separa_palavras(lista_tokens)

# Exibe a quantidade de palavras encontradas apenas com letras na base 'artigos'
print('O número de palavras é', len(lista_palavras))

O número de palavras é 403106


# Processo de normalização

### Retirando palavras repetidas

In [53]:
# função que recebe a 'lista_palavras', trasnforma a palavra para que fique toda minúscula 
def normalizacao(lista_palavras):
    lista_normalizada = []
    for palavra in lista_palavras:
        lista_normalizada.append(palavra.lower())
    return lista_normalizada

In [55]:
lista_normalizada = normalizacao(lista_palavras)
lista_normalizada[:500]

['imagem',
 'temos',
 'a',
 'seguinte',
 'classe',
 'que',
 'representa',
 'um',
 'usuário',
 'no',
 'nosso',
 'sistema',
 'java',
 'para',
 'salvar',
 'um',
 'novo',
 'usuário',
 'várias',
 'validações',
 'são',
 'feitas',
 'como',
 'por',
 'exemplo',
 'ver',
 'se',
 'o',
 'nome',
 'só',
 'contém',
 'letras',
 'o',
 'cpf',
 'só',
 'números',
 'e',
 'ver',
 'se',
 'o',
 'usuário',
 'possui',
 'no',
 'mínimo',
 'anos',
 'veja',
 'o',
 'método',
 'que',
 'faz',
 'essa',
 'validação',
 'java',
 'suponha',
 'agora',
 'que',
 'eu',
 'tenha',
 'outra',
 'classe',
 'a',
 'classe',
 'produto',
 'que',
 'contém',
 'um',
 'atributo',
 'nome',
 'e',
 'eu',
 'quero',
 'fazer',
 'a',
 'mesma',
 'validação',
 'que',
 'fiz',
 'para',
 'o',
 'nome',
 'do',
 'usuário',
 'ver',
 'se',
 'só',
 'contém',
 'letras',
 'e',
 'aí',
 'vou',
 'criar',
 'outro',
 'método',
 'para',
 'fazer',
 'a',
 'mesma',
 'validação',
 'ou',
 'criar',
 'uma',
 'interface',
 'ou',
 'uma',
 'classe',
 'que',
 'tanto',
 'usuario

### Texto normalizado, todas as palavras que contém apenas letras foram separadas em uma lista, e trasnformada em minúscula.

In [57]:
# A funcção set, retorna os valores da lista sem repetições
set([4, 4, 0, 1, 2, 3, 3, 3, 5, 6, 6, 8, 9, 7])

{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}

In [58]:
len(set(lista_normalizada))

18465

### De um total de 403106 palavras, fora reduzido para 18465 palavras sem reperições e sem caracteres